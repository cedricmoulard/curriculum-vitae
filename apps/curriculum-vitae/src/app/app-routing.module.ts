import { NgModule } from "@angular/core";
import { RouterModule, Routes } from "@angular/router";

const appRoutes: Routes = [
  {
    path: "home",
    loadChildren: () => import("./home/home.module").then(mod => mod.HomeModule)
  },
  {
    path: "experiences",
    loadChildren: () =>
      import("./experiences/experiences.module").then(
        mod => mod.ExperiencesModule
      )
  },
  {
    path: "",
    redirectTo: "/home",
    pathMatch: "full"
  }
];

@NgModule({
  imports: [
    RouterModule.forRoot(
      appRoutes,
      { enableTracing: false } // <-- debugging purposes only
    )
  ],
  exports: [RouterModule]
})
export class AppRoutingModule {}
