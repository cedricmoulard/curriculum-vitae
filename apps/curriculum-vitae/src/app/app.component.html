<cv-template-ui [title]="title">
  <div template-content>
    <cv-experience
      *ngFor="let experience of experiences"
      [experience]="experience"
    ></cv-experience>
  </div>
  <div template-sidebar>Sidebar</div>
  <div template-footer>Made with â™¡ by Jeko</div>
</cv-template-ui>
