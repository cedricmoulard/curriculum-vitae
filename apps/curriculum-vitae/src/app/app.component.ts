import { Component } from "@angular/core";
import { experiences } from "./data/experiences";
import { Experience } from "@cv/experience-interface";

@Component({
  selector: "cv-root",
  templateUrl: "./app.component.html",
  styleUrls: ["./app.component.scss"]
})
export class AppComponent {
  title = "CÃ©dric Moulard";
  experiences: Experience[];

  constructor() {
    this.experiences = experiences;
  }
}
