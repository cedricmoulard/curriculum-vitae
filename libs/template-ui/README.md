# template-ui

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test template-ui` to execute the unit tests.
