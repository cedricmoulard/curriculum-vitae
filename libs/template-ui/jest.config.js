module.exports = {
  name: "template-ui",
  preset: "../../jest.config.js",
  coverageDirectory: "../../coverage/libs/template-ui",
  snapshotSerializers: [
    "jest-preset-angular/AngularSnapshotSerializer.js",
    "jest-preset-angular/HTMLCommentSerializer.js"
  ]
};
