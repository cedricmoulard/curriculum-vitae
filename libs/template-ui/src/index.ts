export * from "./lib/template-ui.module";
