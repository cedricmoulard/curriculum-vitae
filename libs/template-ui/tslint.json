{
  "extends": "../../tslint.json",
  "rules": {
    "directive-selector": [true, "attribute", "curriculumVitae", "camelCase"],
    "component-selector": [true, "element", "curriculum-vitae", "kebab-case"]
  }
}
