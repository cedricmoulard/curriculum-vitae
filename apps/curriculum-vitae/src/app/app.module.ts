import { BrowserModule } from "@angular/platform-browser";
import { NgModule } from "@angular/core";

import { AppComponent } from "./app.component";
import { TemplateUiModule } from "@cv/template-ui";
import { ExperienceUiModule } from "@cv/experience-ui";

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, TemplateUiModule, ExperienceUiModule],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule {}
