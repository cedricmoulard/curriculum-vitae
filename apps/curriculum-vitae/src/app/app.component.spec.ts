import { TestBed, async } from "@angular/core/testing";
import { AppComponent } from "./app.component";
import { TemplateUiModule } from "@cv/template-ui";

describe("AppComponent", () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [AppComponent],
      imports: [TemplateUiModule]
    }).compileComponents();
  }));

  it("should create the app", () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app).toBeTruthy();
  });

  it(`should have as title 'Cédric Moulard'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app.title).toEqual("Cédric Moulard");
  });

  it("should render match snapshot", () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.debugElement.nativeElement;
    expect(compiled).toMatchSnapshot();
  });
});
