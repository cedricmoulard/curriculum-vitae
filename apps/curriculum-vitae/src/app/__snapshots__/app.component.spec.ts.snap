// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render match snapshot 1`] = `
<div
  id="root2"
  ng-version="8.0.0"
>
  <cv-template-ui
    ng-reflect-title="Cédric Moulard"
  >
    <div
      class="container"
    >
      <div
        class="header"
      >
        <div>
          Gauche
        </div>
        <div>
          Cédric Moulard
        </div>
        <div>
          Droite
        </div>
      </div>
      <div
        class="sidebar"
      >
        <div
          template-sidebar=""
        >
          Sidebar
        </div>
      </div>
      <div
        class="content"
      >
        <div
          template-content=""
        >
          
          <cv-experience
            ng-reflect-experience="[object Object]"
          >
            <div
              class="flex"
            >
              <div
                class="summary"
              >
                <div
                  class="date"
                >
                  Fev 2018 - Aujourd'hui
                </div>
                <div>
                  Tech Lead Angular
                </div>
                <div>
                  BPCE
                </div>
              </div>
              <div
                class="description"
              >
                <div>
                  Une description
                </div>
              </div>
            </div>
          </cv-experience>
          <cv-experience
            ng-reflect-experience="[object Object]"
          >
            <div
              class="flex"
            >
              <div
                class="summary"
              >
                <div
                  class="date"
                >
                  Oct 2015 - Jan 2019
                </div>
                <div>
                  CEO
                </div>
                <div>
                  On Sight
                </div>
              </div>
              <div
                class="description"
              >
                <div>
                  Une description
                </div>
              </div>
            </div>
          </cv-experience>
        </div>
      </div>
      <div
        class="footer"
      >
        <div
          template-footer=""
        >
          Made with ♡ by Jeko
        </div>
      </div>
    </div>
  </cv-template-ui>
</div>
`;
